drivers/eeprom/eeprom_stm32_L0_L1.h

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 Nick Brassel (tzarc)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/*
+    The size used by the STM32 L0/L1 EEPROM driver.
+*/
+#ifndef STM32_ONBOARD_EEPROM_SIZE
```

Comment:
Just to make sure, this should be defined by the hardware config files for the specific STM chips, correct?

Reply:
In theory, yes, but I couldn't find it. Usually these things are defined by ldscript, and that sort of thing isn't preprocessor-friendly. The other aspect is that the size is large, which triggered the following issue:

When I attempted to test VIA -- if using the full 6kB available on the L082, or the full 16kB on the L152, the bootup time of the device when performing the initial eeconfig/via reset procedure takes a whole bunch of time. Normally that'd be okay, however, the USB endpoint has already been activated, and at least in the case of Windows, because it doesn't respond during the EEPROM reset Windows decided that the device was malfunctioning.

Without VIA the reset is 36 bytes (4-byte aligned extension of the current eeconfig's use of 34 bytes), and the delay is not noticeable.

I'm thinking that perhaps that the USB init and EEPROM reset order should be reversed, assuming it's the wrong way around. Was planning to investigate further today.

Reply:
Did a bit of digging through the ST headers imported by ChibiOS, looks like I can calculate the EEPROM size, but the preprocessor definition names are different between L0 and L1. Why am I not surprised?

Reply:
Because that would be simple. And it's ChibiOS, so nothing can be simple :D

Reply:
> the initial eeconfig/via reset procedure takes upwards of 30 seconds

Worth documenting somewhere so we can set expectations (maybe even as a build warning based on some threshold of `STM32_ONBOARD_EEPROM_SIZE`?)

Reply:
Yeah I'm not entirely sure how I want to handle this.
I'm thinking perhaps we do something like:
```c
#ifndef STM32_ONBOARD_EEPROM_SIZE
#    ifdef VIA_ENABLE
#        define STM32_ONBOARD_EEPROM_SIZE 1024
#    else
#        include "eeconfig.h"
#        define STM32_ONBOARD_EEPROM_SIZE (((EECONFIG_SIZE + 3) / 4) * 4)  // based off eeconfig's current usage, aligned to 4-byte sizes, to deal with LTO and EEPROM page sizing
#    endif
#endif

#if STM32_ONBOARD_EEPROM_SIZE > 128
#    pragma message("Please note: resetting EEPROM using an STM32L0/L1 device takes up to 6.5 seconds for every kB of internal EEPROM used.")
#endif
```

I know the initial thought was to use the total size of the EEPROM available, but I think I'd prefer the defaults to be relatively quick, given that people are unlikely to be configuring more than the usual amounts anyway.

Reply:
.....which is precisely what I've done. Should be right to go if others agree.